Component Library 1 and 2 elements

alexandercyu opened the pull request:
Adds CSS, jinja templates, and javascript for:

- Dropdown lists
- Dropdown panels
- Large nav buttons
- Medium selection buttons
- Small nav buttons
- Regular default buttons
- Alternative button styles for Regular Default and Small Default
- Close buttons
- Popups
- Tags

alexandercyu committed:
Create requirements.txt

alexandercyu committed:
Create build.py

alexandercyu committed:
Create stac-ui.css

- Add legacy design styles
- Add styles for search box

alexandercyu committed:
Create main.html

alexandercyu committed:
Create searchbox.html

Search box partial

alexandercyu committed:
Create sample.js

Add functions for setting and changing placeholder text in search box

alexandercyu committed:
Create ui-styles-scripts.js

Add functions for changing placeholder text in search box

alexandercyu committed:
Create sample.html

Sample page to showcase components

alexandercyu committed:
Move documentation to stac-ui-tools repo

alexandercyu committed:
Update stac-ui.css

Add styles for search box and dropdown

alexandercyu committed:
Create dropdown.html

Add dropdown partial

alexandercyu committed:
Update ui-styles-scripts.js

- Add functions to toggle placeholder text in search box
- Add functions to toggle styles in dropdown

alexandercyu committed:
Update sample.html

Add dropdown to samples

alexandercyu committed:
Update sample.js

Add functions to swap placeholder text

alexandercyu committed:
Update stac-ui.css

Add styles for dropdown, accordion dropdown, and toggle siwtch

alexandercyu committed:
Update main.html

Add requirements for Bootstrap icons, Bootstrap switch

alexandercyu committed:
Create accordion.html

Add accordion partial for accordion dropdown

alexandercyu committed:
Update dropdown.html

Add dropdown partial for regular dropdown and accordion dropdown

alexandercyu committed:
Create toggle.html

Add toggle switch partial

alexandercyu committed:
Update sample.js

Add javascript to toggle dropdown styles

alexandercyu committed:
Update ui-styles-scripts.js

Javascript to toggle dropdown styles

alexandercyu committed:
Update sample.html

Page to showcase UI elements

alexandercyu committed:
Update dropdown.html

Fix dropdown regular dropdown panels

alexandercyu committed:
Update stac-ui.css

Add css for popup, dropdowns, tags

alexandercyu committed:
Update dropdown.html

Add final dropdowns

alexandercyu committed:
Create popup.html

Add popup partial

alexandercyu committed:
Create tag.html

Add tag partial

alexandercyu committed:
Add modal javascript

alexandercyu committed:
Update sample.js

Add dropdown scripts

alexandercyu committed:
Update ui-styles-scripts.js

Add toggle scripts

alexandercyu committed:
Create checkbox.html

Add checkbox partial

alexandercyu committed:
Update sample.html

Add dropdowns, modal, tags to sample

alexandercyu committed:
Create button.html

Add button partial

alexandercyu committed:
Update main.html

Add modal.js to main

alexandercyu committed:
Update popup.html

Fix dictionary and output of dictionary values

alexandercyu committed:
Update stac-ui.css

- Fix shadow and border of popup
-align items in popup

alexandercyu committed:
Update button.html

Add reg-default and reg-default-alt buttons

alexandercyu committed:
Update stac-ui.css

Reuse CSS for eg-default and reg-default-alt buttons

alexandercyu committed:
Update sample.html

Add reg-default and reg-default-alt buttons

alexandercyu committed:
Update button.html

Added Next and Descending buttons

alexandercyu committed:
Update stac-ui.css

Added Next and Descending button css

alexandercyu committed:
Update popup.html

- Fixed array
- Moved close button out of content heading

alexandercyu committed:
Update sample.html

Added next and descending buttons

alexandercyu committed:
Update button.html

Update Ascending and Descending icons

alexandercyu committed:
Update dropdown.html

Add Frequency dropdown

alexandercyu committed:
Update sample.html

Add Frequency dropdown

alexandercyu committed:
Update stac-ui.css

Add css for Frequency dropdown

mishaschwartz reviewed (commented):
Please re-request a review once library 2 has been added as well

ShrutiK100 reviewed (changes requested):
The search bar will have a search button on its left (26px from the right side border of the search bar). This button has been added and mentioned in a comment on Figma.

Font colour change (search bar): When a user starts typing, the font colour should change to black. Currently, the font colour is grey throughout the search process

The default regular dropdown height is 64 and the width is 172. Please check the button size.

Advanced filters such as frequency, location, and time frame do not have a chevron with their list items.  
<img width="441" alt="Screenshot 2024-12-06 at 3 04 07 PM" src="https://github.com/user-attachments/assets/0975b469-6fd4-43e0-85ba-4e2cde9d8e66">

alexandercyu committed:
Update main.html

Add main.js

alexandercyu committed:
Update button.html

- Add spin button
- Add conditions so other button layouts don't load when spin button is wanted

alexandercyu committed:
Create main.js

Add functions to increment and decrement number in textbox

alexandercyu committed:
Update sample.js

- Add listeners for increment and decrement
- Set placeholder text for small number input
- set placeholder for date input

alexandercyu committed:
Update ui-styles-scripts.js

Add functions to toggle styles for small number input

alexandercyu committed:
Update sample.html

- Add "Does Not Repeat" dropdown
- add small number input
- add spin button
- add date input
- add time input

alexandercyu committed:
Update button.html

Add radio button

alexandercyu committed:
Create info-display.html

Add partial for displaying search results in row format and card format

alexandercyu committed:
Create radio.html

Add radio button

alexandercyu committed:
Update stac-ui.css

- Add css for info display in row and card
- Add css for tags in info display card format
- Add css for radio button
- Add css for small text/number input field
- Add css for spin button
- Add css for date and time input fields (Note: these may be removed later)

alexandercyu committed:
Update tag.html

- Collapse default tag options into array instead of dict
- Add option for swapping style classes for tag row and tags in cards

alexandercyu committed:
Create textbox.html

- Add date and time fields (Note: these may be removed later)

alexandercyu committed:
Update sample.html

Add medium dropdown, small dropdown, radio button, info display in row and card

alexandercyu committed:
Update dropdown.html

Use bootstrap dropdown instead

alexandercyu committed:
Update stac-ui.css

Add css to override bootstrap dropdown

alexandercyu committed:
Update dropdown.html

Change dropdown medium list and dropdown small list into bootstrap dropdowns

alexandercyu committed:
Update stac-ui.css

- Override bootstrap css for dropdown medium list and dropdown small list
- Add styles for dropdown medium list and dropdown small list
- remove old styles for dropdown medium list

alexandercyu committed:
Create datepicker.html

Add datepicker - input box with jquery calendar

alexandercyu committed:
Update stac-ui.css

- Add css for jquery datepicker
- comment out css for input-date-reg for now

alexandercyu committed:
Create datepicker.js

Add javascript to initialize date picker

alexandercyu committed:
Update main.html

Add date picker jquery and default css

alexandercyu committed:
Update stac-ui.css

Clean up commented styles and spacing

alexandercyu committed:
Rename popup.html to modal-metadata.html

alexandercyu committed:
Update datepicker.html

Add datepicker using flatpickr

alexandercyu committed:
Create timepicker.html

Add time picker using flatpickr

alexandercyu committed:
Delete datepicker.js

Have script directly in datepicker.html

alexandercyu committed:
Update stac-ui.css

Fix icon on dropdown timeframe

alexandercyu committed:
Update dropdown.html

Fix icon on dropdown timeframe

alexandercyu committed:
Update datepicker.html

update label

alexandercyu committed:
Update timepicker.html

Update label

alexandercyu committed:
Update textbox.html

Remove input date and input time

alexandercyu committed:
Create modal-recurrence.html

Add recurrence modal

alexandercyu committed:
Update button.html

Remove unused items in dictionary

alexandercyu committed:
Update main.html

Remove jquery date picker

alexandercyu committed:
Update checkbox.html

Update label alignment

alexandercyu committed:
Update info-display.html

Update alignment

alexandercyu committed:
Delete toggle.html

alexandercyu committed:
Create modal-recurrence.js

Add script for modal recurrence

alexandercyu committed:
Delete main.js

alexandercyu committed:
Update modal-recurrence.js

Remove unused function

alexandercyu committed:
Update sample.js

Remove unused function

alexandercyu committed:
Create spinbutton.js

alexandercyu committed:
Update ui-styles-scripts.js

Remove unused function

alexandercyu committed:
Update sample.html

Add samples fro date picker and time picker

alexandercyu commented:
Component library 2 elements added

alexandercyu committed:
Merge branch 'main' into ui-components

mishaschwartz commented on templates/partials/searchbox.html:
```diff
@@ -0,0 +1,8 @@
+
+    <form role="search">
+        <input id="searchInput" type="search" class="search-input" />
+        <i class="fa-solid fa-magnifying-glass search-icon"></i>
```
This needs to be vertically centred and needs a white circular background

alexandercyu replied:
Background added to icon, and icon centered.

mishaschwartz commented on templates/partials/dropdown.html:
This entire file is too complex. Please refactor it.

If there is truly no overlap between the dropdown types, move each to a separate file. 

But I really think all this can be refactored so the dropdown types can share a lot of code.

alexandercyu replied:
Separated out panel type dropdowns from list types.

List types combined to share code.

mishaschwartz commented on templates/partials/dropdown.html:
```diff
@@ -0,0 +1,392 @@
+
+            <ul id="{{ dropdown_list_id }}" class="dropdown-menu {{ dropdown_library[dropdown_type]["list_class"] }}">
+                {% for item in list_items %}
+                    {% set list_item_id = dropdown_id ~ loop.index|string %}
```
`~` already converts operands to strings. You don't need the explicit cast to string

alexandercyu replied:
Removed the string cast.

mishaschwartz commented on templates/partials/dropdown.html:
```diff
@@ -0,0 +1,392 @@
+            <ul id="{{ dropdown_list_id }}" class="dropdown-menu {{ dropdown_library[dropdown_type]["list_class"] }}">
+                {% for item in list_items %}
+                    {% set list_item_id = dropdown_id ~ loop.index|string %}
+                    <li><a id="{{ list_item_id }}" class="dropdown-item {{ dropdown_library[dropdown_type]["list_item_class"] }} caption" href="#">{{ item }}</a></li>
```
Selecting an item does not show the selected item when the dropdown closes

alexandercyu replied:
Added javascript so small and medium dropdown lists show the selected item when clicked.

mishaschwartz commented on templates/partials/modal-metadata.html:
```diff
@@ -0,0 +1,93 @@
+
+
+
+<dialog id="metadataModal" class="dialog-metadata">
```
dialog needs to horizontally scrollable for small screens

mishaschwartz commented on templates/partials/modal-metadata.html:
```diff
@@ -0,0 +1,93 @@
+
+<dialog id="metadataModal" class="dialog-metadata">
+    <div class="popup-metadata-container">
+        <div class="popup-button-close">
```
X is not centred in box

alexandercyu replied:
Fixed alignment so X is centered in box.

mishaschwartz commented on templates/partials/dropdown.html:
```diff
@@ -0,0 +1,392 @@
+{% endif %}
+
+
+{% if dropdown_type == "regular_dropdown_list_metadata" or dropdown_type == "regular_dropdown_list_license" or dropdown_type == "regular_dropdown_list_frequency"%}
```
Hover colours are not consistent with what is shown in the design document in Figma

alexandercyu replied:
The colour applied is what is noted in the design:

background: rgba(136, 144, 194, 0.25);

It has an opacity and the design has a grey background for those elements.  Perhaps that is causing the difference?

You can see in the screenshot the shape menu items have the same colour fill as the background of the dropdown elements, but the shape menu has a white background.

![Screenshot 2024-12-19 at 11 57 54 AM (3)](https://github.com/user-attachments/assets/4ab0eaec-5700-4723-ae42-10ae1cae6e7e)

mishaschwartz replied:
As described in library 2:

- if not expanded: grey background
- if expanded and no hover: white background
- if expanded and hover: grey background

mishaschwartz replied:
This is still not consistent with what it described in Figma

alexandercyu replied:
I left a note in Figma asking Shruti about the background colours.  @ShrutiK100 , could you take a look?

mishaschwartz commented on templates/partials/dropdown.html:
```diff
@@ -0,0 +1,392 @@
+
+
+{% if dropdown_type == "small_dropdown_list" or dropdown_type == "medium_dropdown_list" %}
+    <div class="{{ dropdown_library[dropdown_type]["container_class"] }}">
```
chevron is not vertically centred

alexandercyu replied:
CSS added to vertically centre the chevron but it seems to be centred relative to the parent element and not aligned with the text.  Will try changing the size of the chevron to see if it will work.

mishaschwartz commented on templates/partials/button.html:
This needs to be refactored as well. If the button types have nothing in common their code should go in separate files.

alexandercyu replied:
Refactored so spin button  and radio button are in their own files.

mishaschwartz commented on templates/partials/modal-recurrence.html:
this modal can't be closed

alexandercyu replied:
Added close button

mishaschwartz commented on templates/partials/tag.html:
```diff
@@ -0,0 +1,19 @@
+
+<div class="{{ tag_input["container_class"] }}">
+    {% for tag in tag_input["tag_array"] %}
+        <div class="{{ tag_input["class"] }}">{{ tag }}</div>
```
These need to be buttons and/or clickable

alexandercyu replied:
Changed to links.  Currently just anchor links with name of tag.

mishaschwartz commented on templates/site/sample.html:
```diff
@@ -0,0 +1,302 @@
+    <br>
+    <br>
+
+    {% with %}
```
This is part of a table. Please create the table component as well and show it with multiple rows

mishaschwartz replied:
this still needs to be done too

alexandercyu replied:
Example made to show multiple rows in a table.

NOTE: Example shows table tag in sample.html.  In production the table tag will be moved to the template and a for loop will be added to loop through passed table values.

mishaschwartz commented on templates/site/sample.html:
```diff
@@ -0,0 +1,302 @@
+        <br>
+    <br>
+
+    {% with %}
```
This is part of a grid. Please create the grid component as well and show it with multiple cards

mishaschwartz replied:
This still needs to be done

alexandercyu replied:
Example made to show multiple cards.

mishaschwartz commented on templates/partials/info-display.html:
```diff
@@ -0,0 +1,59 @@
+            {% set tag_input_custom ={
+            "container_class": "tag-info-card-container",
+            "class": "tag-info-card",
+            "tag_array": ["XML", "HTML", "NetCDF"]
```
tag values should clickable

alexandercyu replied:
Made tag values clickable.  Also changed highlight style of tag when hovering mouse over them.

mishaschwartz commented on static/css/stac-ui.css:
```diff
@@ -0,0 +1,1796 @@
+    border-color: var(--indigo250-opacity-light);
+    border-radius: 9px;
+}
+/*
```
remove commented code

alexandercyu replied:
Sorry, this is for adjusting the arrow buttons in the calendar month menu.  I was having trouble getting the spacing the way Shruti has in the design.

Added a comment for this.   

Hopefully the default position fits with Shruti's design and this part can be deleted.

mishaschwartz reviewed (changes requested):
This is a good start!

Please see the comments below. 

There a few places that the visual is not consistent with what is in Figma. Please address those issues first.

There are two files that need to be refactored (see comments). If you would like to discuss this, please let me know. Especially if my comments are not clear.

Finally, there are a lot of hardcoded examples in this which is fine to demonstrate what the components look like. However, I'm concerned that the way you've written the code will make it difficult to dynamically add in the values later on. You don't need to change anything for this PR but please start thinking about how dynamic values can be inserted and how to refactor the code in a later PR to accommodate this.

alexandercyu committed:
Update dropdown.html

Removed cast to string as per feedback

alexandercyu committed:
Update searchbox.html

Added white background to magnifying glass icon as per updated design

alexandercyu committed:
Update stac-ui.css

- Added styles for  white background to magnifying glass icon as per updated design
- Fixed alignment of 'X' icon for close button
- Added comment for css to adjust alignment of arrow buttons in calendar menu

alexandercyu committed:
Merge branch 'ui-components' of https://github.com/DACCS-Climate/stac-search-ui into ui-components

alexandercyu commented:
> The search bar will have a search button on its left (26px from the right side border of the search bar). This button has been added and mentioned in a comment on Figma.
> 
> Font colour change (search bar): When a user starts typing, the font colour should change to black. Currently, the font colour is grey throughout the search process
> 
> The default regular dropdown height is 64 and the width is 172. Please check the button size.
> 
> Advanced filters such as frequency, location, and time frame do not have a chevron with their list items. <img alt="Screenshot 2024-12-06 at 3 04 07 PM" width="441" src="https://private-user-images.githubusercontent.com/161752175/393389477-0975b469-6fd4-43e0-85ba-4e2cde9d8e66.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3MzA1NjgsIm5iZiI6MTczNDczMDI2OCwicGF0aCI6Ii8xNjE3NTIxNzUvMzkzMzg5NDc3LTA5NzViNDY5LTZmZDQtNDNlMC04NWJhLTRlMmNkZTlkOGU2Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyMFQyMTMxMDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZWZiZDIwMGZjNTM3NDVmYTM5ZTkwOWM3ODRkMTgzZTAwNGJjMThiMmNkZDM0OWI2NmViMjhkMGRhNTg4MWM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.GtdN8XHtsNSxz8HdE-_ThNWPHobT1HKERMNxQmCCrMI">

Font colour when typing is now black.

alexandercyu committed:
Create dropdown-panel.html

Split off dropdown panels to separate file

alexandercyu committed:
Update stac-ui.css

Added styles for dropdown panel

alexandercyu committed:
Update ui-styles-scripts.js

Added scripts to toggle dropdown panel styles

alexandercyu committed:
Update sample.html

Added dropdown panel

alexandercyu committed:
Update stac-ui.css

- Added styles to override Bootstrap styles for dropdowns
- Overrode position for timeframe panel so the containing Bootstrap div expands over entire panel
- Use Bootstrap CSS icon for chevron icon instead of HTML Bootstrap tag
- Get rotating chevron working

alexandercyu committed:
Update dropdown-panel.html

- Redo Share dropdown using Bootstrap elements
- Fix border of Timeframe dropdown
- Need to fix hover colour of Timeframe dropdown

alexandercyu committed:
Update ui-styles-scripts.js

Changed scripts involved in adding classes for rotating chevron

alexandercyu committed:
Update dropdown-panel.html

Separate panel type dropdowns into another file
- remove list type dropdowns
- get chevron rotations working with css
- add functionality to toggle title text for Source dropdown to display STAC version (NOTE: this does not work from localhost.  This is coded to work using the Magpie API)

alexandercyu committed:
Update dropdown.html

Separate panel type dropdowns into another file

alexandercyu committed:
Update ui-styles-scripts.js

- Modify chevron rotation functions to use the supplied element ID and css class
- Add function to toggle text in Source dropdown and display stac version.  Note this does not work on localhost and uses the Magpie API

alexandercyu committed:
Update sample.html

Update samples displayed

alexandercyu committed:
Update dropdown.html

- Get colour swapping of default dropdown list working and rotation of chevron working in default dropdown list
- Integrate metadata dropdown list into same template as default dropdown list.
-TODO: fix chevron rotating in metadata dropdown
: find where border radius is set and remove
: find where spacing of chevron is set and fix

alexandercyu committed:
Update stac-ui.css

Update styles to get colour swapping right on default dropdown list

alexandercyu committed:
Update sample.js

Get scripts working for swapping colours and chevron rotating

alexandercyu committed:
Update dropdown-panel.html

Change function name

alexandercyu committed:
Update ui-styles-scripts.js

Add functions for text colour swap

alexandercyu committed:
Update dropdown.html

Refactored default dropdown list, metadata dropdown list, frequency dropdown list , and license frequency dropdown list to use same template with variations

alexandercyu committed:
Update stac-ui.css

- Removed css for dropdowns that used checkbox states to expand the list.  Instead uses Bootstrap dropdown template
- Changed chevron icons in metadata dropdown list, frequency dropdown list, license dropdown list and default dropdown list so all use bootstrap icon and css to rotate

alexandercyu commented:
> he default regular dropdown height is 64 and the width is 172. Please check the button size.
> 
> Advanced filters such as frequency, location, and time frame do not have a chevron with their list items.

- Fixed the height of the regular dropdown 
- Removed chevron from frequency, license dropdown list items

alexandercyu committed:
Update stac-ui.css

- changed tag css so the background of each tag is highlighted instead of the entire row
- made each tag a link and added styles to change font colour and cursor
- in card view, changed tag background colour
- moved margin to tag css class
- removed set width from tag container
- centered chevron in dropdown-small-list and  dropdown-medium-list (may need font size changed.  See if possible.)

alexandercyu committed:
Create button-radio.html

Split off radio button to it's own file

alexandercyu committed:
Create button-spin.html

Split off spin button to its own file

alexandercyu committed:
Update button.html

Removed spin button and radio button from button file

alexandercyu committed:
Delete radio.html

Renamed file to button-radio.html

alexandercyu committed:
Update tag.html

Made changes to row style and panel style tag arrangement

alexandercyu committed:
Update modal-recurrence.js

- removed spin button functions.  Spin button functions are in their own template

alexandercyu committed:
Create ui-function-scripts.js

Used to prevent button clicks in timeframe panel from bubbling up

alexandercyu committed:
Update sample.js

- moved dropdown script to dropdown.html template
- moved spin button script to spin button template

alexandercyu committed:
Update sample.html

- remove unecessary settings

alexandercyu committed:
Update modal-recurrence.html

Add close button

alexandercyu committed:
Update dropdown.html

- removed unneeded settings
- moved panel type dropdowns to their own template

alexandercyu committed:
Update dropdown-panel.html

- Change radio button template call to match file name
- make timeframe panel dropdown chevron rotate

alexandercyu committed:
Update info-display.html

- Remove hard coded example tags
 - Update template call to reflect changes in tag template

alexandercyu committed:
Update checkbox.html

Remove margins

alexandercyu committed:
Update dropdown.html

Add function to swap or replace values on list item click event

alexandercyu committed:
Create dropdown-list.js

Create function to swap or replace values on list item click event

alexandercyu committed:
Update sample.html

Add dropdown_id and label_id attributes to template

mishaschwartz commented on templates/partials/dropdown.html:
```diff
@@ -0,0 +1,225 @@
+        </div>
+    {% if dropdown_type == "default_dropdown_list" %}
+        <script>
+            document.addEventListener("DOMContentLoaded", function () {
```
This is much nicer but still not quite as expected:

- clicking the drop down from the default state rotates the chevron up :+1: 
- clicking the drop down from the expanded state rotates the chevron down :+1: 
- clicking an element in the drop down rotates the chevron down without collapsing the drop down :-1: 
- clicking outside of the drop down in the expanded state hides the drop down but doesn't rotate the chevron and the background goes to white :-1: 

What should happen:

- if the drop down is collapsed: chevron should be down
- if the drop down is expanded: chevron should be up

It shouldn't matter how the drop down is expanded/collapsed, the chevron direction should be consistent

mishaschwartz replied:
Also the background colour when expanded is still not consistent with what's on Figma

alexandercyu replied:
Made issue for this

mishaschwartz commented on templates/partials/dropdown.html:
```diff
@@ -0,0 +1,225 @@
+                        </script>
+                    {% endif %}
+
+                    {% if dropdown_type == "medium_dropdown_list" %}
```
The height of the initial div should stay the same when expanded:

![image](https://github.com/user-attachments/assets/14e4cee4-8b89-4f81-bde4-cbdeff1cf90f)

The "does not repeat section" should be the same height as above when expanded:
![image](https://github.com/user-attachments/assets/bbccaf76-02ac-4256-bf7d-095360799053)

alexandercyu replied:
"does not repeat section"  now maintains height when list is expanded

mishaschwartz commented on templates/partials/dropdown.html:
```diff
@@ -0,0 +1,225 @@
+                {% for item in list_items %}
+                    {% set list_item_id = dropdown_id ~ loop.index %}
+                    <li><a id="{{ list_item_id }}" class="dropdown-item {{ dropdown_library[dropdown_type]["list_item_class"] }} caption" role="button">{{ item }}</a></li>
+                    {% if dropdown_type == "small_dropdown_list" %}
```
The small drop down list should expand **down** like the rest of the drop down lists. Right now it expands up and down

alexandercyu replied:
Small dropdown now expands like the larger one.

mishaschwartz commented on templates/partials/dropdown.html:
javascript should be in a separate file. Please don't put it inline. It's much easier to understand/review if we don't have multiple syntaxes in a single file unless absolutely necessary

alexandercyu replied:
I originally wanted to pass the jinja variables into the javascript so the element ID passed will be dynamic.

I moved the scripts to their own file for dropdowns that I think will only need an ID specified for finding the element and not for a functionality.  I kept the scripts for small and medium dropdown lists since they will need to stay there for swapping the list items so they are visible on click.

mishaschwartz replied:
Ok, that's fine

mishaschwartz commented on templates/partials/modal-recurrence.html:
```diff
@@ -0,0 +1,60 @@
+                {% set target_id = "inputNumberRecurrence" %}
+                {% include "partials/button-spin.html" %}
+            {% endwith %}
+
```
needs more space between these two elements

alexandercyu replied:
Added spacing between the "Repeat Every" input elements.

mishaschwartz commented on templates/partials/button-spin.html:
```diff
@@ -0,0 +1,21 @@
+    <div id="{{ spin_button_up_id }}" class="spin-button-sm-up">
+        <i class="fa-solid fa-play spin-button-sm-up-icon"></i>
+    </div>
+    <div id="{{ spin_button_down_id }}" class="spin-button-sm-dn">
```
hover state should change the background of the whole div, not just 70% of it. (My guess is that the div isn't covering the same space on the screen as the parent)

alexandercyu replied:
Hover now changes whole div background colour.

mishaschwartz commented on templates/partials/button.html:
```diff
@@ -0,0 +1,74 @@
+        "button_sm_default_copy": {
+            "class":"button-sm-default button-sm-default-copy subtitle-2"
+        },
+        "close_button_x": {
```
This button loses the background colour when it loses focus. Is that intended?

alexandercyu replied:
Fixed the actions that control the background colour.

mishaschwartz commented on templates/partials/dropdown-panel.html:
```diff
@@ -0,0 +1,246 @@
+
+                        </div>
+                        <div>
+                           {% with %}
```
These don't behave like radio buttons. Radio buttons should let the use make an **exclusive choice** between multiple options. This does not do this. Please review the code here

alexandercyu replied:
Radio buttons now offer exclusive choice.

mishaschwartz replied:
- Never is the default
- calendar should only be editable if "Ends On" is selected

alexandercyu replied:
- Never radio button made selected by default
- Ends On calender only editable if Ends On radio selected

ShrutiK100 reviewed (changes requested):
<img width="978" alt="Screenshot 2025-01-13 at 11 02 13 AM" src="https://github.com/user-attachments/assets/f2e23cae-447d-496c-92e3-e425916a3418" />

Dropdown options don't match the design
<img width="181" alt="Screenshot 2025-01-13 at 11 08 13 AM" src="https://github.com/user-attachments/assets/7edf1336-eab0-45a3-981d-f4ad233bad5a" />
<img width="366" alt="Screenshot 2025-01-13 at 11 07 54 AM" src="https://github.com/user-attachments/assets/e3e3a4a2-730d-4010-8611-c8b321cac396" />
Also, check the hover properties for each dropdown item. The respective background should change when hovered. It also needs to be set as a selected item once clicked.
<img width="184" alt="Screenshot 2025-01-13 at 11 06 38 AM" src="https://github.com/user-attachments/assets/3a212c9d-c0c3-4f7f-83de-dec1b66ea385" />
<img width="168" alt="Screenshot 2025-01-13 at 11 07 01 AM" src="https://github.com/user-attachments/assets/de6337ef-1947-43e5-b58c-c6c1fcb7333a" />
<img width="427" alt="Screenshot 2025-01-13 at 11 14 24 AM" src="https://github.com/user-attachments/assets/1f1799f0-cd13-4b32-a17d-5beae7f3c4ce" />
<img width="1146" alt="Screenshot 2025-01-13 at 11 18 47 AM" src="https://github.com/user-attachments/assets/1aa1d0b2-c2cc-4d39-b154-cbf91859dc41" />
<img width="852" alt="Screenshot 2025-01-13 at 11 26 13 AM" src="https://github.com/user-attachments/assets/93ae40db-c22f-4717-8a0d-0744113e9150" />
<img width="333" alt="Screenshot 2025-01-13 at 11 28 34 AM" src="https://github.com/user-attachments/assets/7770f527-f71d-4bff-b079-c321efb6b901" />
<img width="317" alt="Screenshot 2025-01-13 at 11 31 35 AM" src="https://github.com/user-attachments/assets/2b24539d-1c8f-48fb-944e-72e3739ec41a" />
<img width="44" alt="Screenshot 2025-01-13 at 11 36 58 AM" src="https://github.com/user-attachments/assets/83b3b1a0-842d-452e-8c22-e517f6db9581" />
<img width="26" alt="Screenshot 2025-01-13 at 11 37 26 AM" src="https://github.com/user-attachments/assets/e73a206e-1aff-49a8-b237-def6c1bedafc" /
Component design doesn't match 

<img width="155" alt="Screenshot 2025-01-13 at 11 41 41 AM" src="https://github.com/user-attachments/assets/ff32d8d6-9514-49de-8475-5d6aca413ad2" />
This button component items need to be center-aligned, 
<img width="145" alt="Screenshot 2025-01-13 at 11 45 18 AM" src="https://github.com/user-attachments/assets/a5090783-be91-4783-aaea-d9c81aaee92d" />


Hover properties missing for "X" button
<img width="113" alt="Screenshot 2025-01-13 at 11 47 45 AM" src="https://github.com/user-attachments/assets/ec9b8b98-9b36-4941-ab72-cdc7300344ac" />
<img width="77" alt="Screenshot 2025-01-13 at 11 47 57 AM" src="https://github.com/user-attachments/assets/6a33bd62-c9fd-4793-8535-c319bf7bd4f9" />

Radiobutton deselect property missing

The content of text input must be center-aligned, DD is sliced
<img width="147" alt="Screenshot 2025-01-13 at 11 50 42 AM" src="https://github.com/user-attachments/assets/ba143dd1-526e-4a95-b166-32940601e7e9" />


<img width="338" alt="Screenshot 2025-01-13 at 11 52 54 AM" src="https://github.com/user-attachments/assets/b89df00d-336f-4e3b-ad2d-b081679bf9d7" />

<img width="262" alt="Screenshot 2025-01-13 at 11 58 50 AM" src="https://github.com/user-attachments/assets/71d8c56f-a000-4b7a-9c10-d23693213c9a" />

<img width="246" alt="Screenshot 2025-01-13 at 11 57 50 AM" src="https://github.com/user-attachments/assets/66b4fc7d-03d8-46e1-974a-f633f98c3dc8" />

alexandercyu commented:
> It also needs to be set as a selected item once clicked.

What will happen if multiple items are checked?  What will be set in that use case?

>Radiobutton deselect property missing

Radio buttons are used for discrete choices., they are not deselected.  If you want a deselect to be used, checkboxes should be used instead.

mishaschwartz commented:
>> It also needs to be set as a selected item once clicked.

> What will happen if multiple items are checked? What will be set in that use case?

It will be set to all items that are selected. How that is displayed to the user is up to you, @alexandercyu and @ShrutiK100.

@ShrutiK100 please add to Figma a demo of what the checkbox looks like if one option is selected and if multiple are selected. 

>>  Radiobutton deselect property missing

> Radio buttons are used for discrete choices., they are not deselected. If you want a deselect to be used, checkboxes should be used instead.

@alexandercyu right now the radio buttons cannot be used to make discrete choices. Selecting one does not deselect the other.

ShrutiK100 commented:
> > > It also needs to be set as a selected item once clicked.
> 
> > What will happen if multiple items are checked? What will be set in that use case?
> 
> It will be set to all items that are selected. How that is displayed to the user is up to you, @alexandercyu and @ShrutiK100.
> 
> @ShrutiK100 please add to Figma a demo of what the checkbox looks like if one option is selected and if multiple are selected.
> 
> > > Radiobutton deselect property missing
> 
> > Radio buttons are used for discrete choices., they are not deselected. If you want a deselect to be used, checkboxes should be used instead.
> 
> @alexandercyu right now the radio buttons cannot be used to make discrete choices. Selecting one does not deselect the other.
@alexandercyu @mishaschwartz Thanks for your input.
The best practice for such a dropdown would be: 
Multiple selection: To display all selected options 
Single selection: Display the chosen option only
All selection: Display text- "All Catalogs"

Refer for Multiple selection : 
<img width="723" alt="Screenshot 2025-01-14 at 10 13 58 AM" src="https://github.com/user-attachments/assets/dcab5cbe-ed50-49ee-8f52-69ef168e5f92" />

@mishaschwartz Let me know if this display option works I will re-design the component

mishaschwartz commented:
@ShrutiK100 looks good to me

alexandercyu commented:
>Hover properties missing for "X" button

Hover properties were not defined in the design.  Only default and pressed.

I'll add hover.

alexandercyu commented:
> @alexandercyu right now the radio buttons cannot be used to make discrete choices. Selecting one does not deselect the other.

Radio buttons now offer discrete choices.

alexandercyu commented:
> small number input in recurrence panel

The design shows the background changes when the cursor is in the box (when the box is pressed/clicked). That is what happens.

alexandercyu committed:
Update button-radio.html

Added css to align radio and label on same row

alexandercyu committed:
Update dropdown-panel.html

- Added script to set Never radio as selected and Ends On calendar disabled by default
- Added script to enable Ends On calendar when Ends On radio selected
- Added script to disable Ends On calendar when Never radio selected

alexandercyu committed:
Create dropdown-timeframe-scripts.js

- Added script to set Never radio as selected and Ends On calendar disabled by default
- Added script to enable Ends On calendar when Ends On radio selected
- Added script to disable Ends On calendar when Never radio selected

alexandercyu commented:
> The content of text input must be center-aligned, DD is sliced

Placeholder text no longer cut off.

Spacing added between title and input field.

alexandercyu commented:
<img alt="Screenshot 2025-01-13 at 11 52 54 AM" width="338" src="https://private-user-images.githubusercontent.com/161752175/402627095-b89df00d-336f-4e3b-ad2d-b081679bf9d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcwNDUwNDIsIm5iZiI6MTczNzA0NDc0MiwicGF0aCI6Ii8xNjE3NTIxNzUvNDAyNjI3MDk1LWI4OWRmMDBkLTMzNmYtNGUzYi1hZDJkLWIwODE2NzliZjlkNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNlQxNjI1NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02YzRmMGFjMzg4YmEyYmJkNWE4MTQ0MmJiOTA3Y2MxNDYzNjQ2YjM4MDk3MjgxYmFkZjg5ZTNhODJlYzFiOTY5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ZimlGS_P0udct0KWocnUIM-xxHIA5gp_0Y4SBdSiHg8"> 

- Font for "Custom Recurrence" and "Repeat Every" fixed.
- Spacing in"Repeat Every" elements fixed
- Background colour for number field now only grey when there is no value inside it.

mishaschwartz commented:
> TODO: close modal when clicking outside of it

This isn't necessary

This guy has a good summary of why: https://www.reddit.com/r/webdev/comments/19fg8a9/comment/kjjxtma/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

Also, here are some (very in depth) recommendations for best-practices: https://html.spec.whatwg.org/multipage/interaction.html#close-requests-and-close-watchers

alexandercyu commented:
<img width="26" alt="Screenshot 2025-01-13 at 11 37 26 AM" src="https://github.com/user-attachments/assets/e73a206e-1aff-49a8-b237-def6c1bedafc" / Component design doesn't match
>

- Using svgs from figma now.
- Fixed hover

alexandercyu commented:
<img alt="Screenshot 2025-01-13 at 11 41 41 AM" width="155" src="https://private-user-images.githubusercontent.com/161752175/402622062-ff32d8d6-9514-49de-8475-5d6aca413ad2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcxMjI1MDUsIm5iZiI6MTczNzEyMjIwNSwicGF0aCI6Ii8xNjE3NTIxNzUvNDAyNjIyMDYyLWZmMzJkOGQ2LTk1MTQtNDlkZS04NDc1LTVkNmFjYTQxM2FkMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExN1QxMzU2NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMzU5OWMxZjJlOGZlMDNkZDA2MTdlYzI2MTM0MGZmZjBkNWZhZjA1MmI1MTE2OWYwMjE2Mjk3ZDkxODA3ZTEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6WRou3hPvaRFp1GqGeG4UtJDa6VnlJ4nNEA8tTUXPe8"> This button component items need to be center-aligned, <img alt="Screenshot 2025-01-13 at 11 45 18 AM" width="145" src="https://private-user-images.githubusercontent.com/161752175/402623490-a5090783-be91-4783-aaea-d9c81aaee92d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcxMjI1MDUsIm5iZiI6MTczNzEyMjIwNSwicGF0aCI6Ii8xNjE3NTIxNzUvNDAyNjIzNDkwLWE1MDkwNzgzLWJlOTEtNDc4My1hYWVhLWQ5YzgxYWFlZTkyZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExN1QxMzU2NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZjgzNmQzZDczNzRhMzcxMTJkNjI0YWRkMDEwMmQ5MGExMDI0NzQ4MTFiOGM1YzBmOTkxZmZlMjQzY2Y4MDVhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.trgj75QtKUuPNrpfeUCG8rhFj2BJWRzGqVOrLAbQc3Q">
> 

Component items already centre aligned.  Added space between icon and text.

alexandercyu committed:
Update dropdown.html

- Fixed function to  swap of bottom border radius in first list item in medium dropdown list
- Rename list items in default dropdown list
- remove data-bs-auto-close="outside" bootstrap setting the forces dropdown to only close from clicking outside it
- change anchor tag in dropdown regular to include role="button" so it will stop going to the top of the page on click
- add script to replace the first list item instead of swapping the list items

alexandercyu committed:
Update stac-ui.css

- Fix font of date input field label to capitalize first letter
- add hover style for close button
- add margins for previous and next button icons to increase space between icon and button text
-add css to align button text in regular list items
- override margin spacing in regular list ul
- remove background colour unset in bootstrap btn class
- fix height in first list item in default list button (small and  medium list types)
- fix border radius corners not swapping in medium list first item (NOTE: Last item still needs to be fixed)
- use custom svg instead of fontawesome icons for spin button
- fix margins for spin button icons
- fix alignment of spin button icons
- add colour swap for svg spin button icons
- add mirgins for date picker and time picker labels
- add hover styles for table row for displaying collection info as a row
- add styles for displaying collection info as cards when there are multiple cards in a row

alexandercyu committed:
Update button-spin.html

- Add custom svg icons for spin button icons instead of using fontawesome
- target small input field background now white when there is value in field

alexandercyu committed:
Update button.html

remove default padding in buttons

alexandercyu committed:
Update info-display.html

Add table rows for sample display

alexandercyu committed:
Update modal-recurrence.html

Add text font class

alexandercyu committed:
Update modal-recurrence.js

Background colour of number input remains white if there are values in it

alexandercyu committed:
Update spinbutton.js

background colour of number input remains white when there are values in it

alexandercyu committed:
Update sample.html

- Add ids for dropdown elements
- change dropdown titles to match changes
- Add table for  example display only

alexandercyu committed:
Update stac-ui.css

-modify css to see if it will put rounded corners on last list item in medium dropdown list
- remove gray background in small number input (background is now always white)

alexandercyu committed:
Update modal-recurrence.js

Remove function call to change background in small number input

alexandercyu committed:
Update spinbutton.js

Remove function to change background in small number input

alexandercyu committed:
Update sample.html

Add identifier to card type display of information

alexandercyu committed:
Update button-spin.html

Remove change background colour functions

alexandercyu committed:
Update stac-ui.css

Fixed width in default dropdown so it changes according to the text length

alexandercyu committed:
Update stac-ui.css

Added blank space to bottom of metadata modal

alexandercyu commented:
<img alt="Screenshot 2025-01-13 at 11 18 47 AM" width="1146" src="https://private-user-images.githubusercontent.com/161752175/402616066-1aa1d0b2-c2cc-4d39-b154-cbf91859dc41.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc1Nzk4NTIsIm5iZiI6MTczNzU3OTU1MiwicGF0aCI6Ii8xNjE3NTIxNzUvNDAyNjE2MDY2LTFhYTFkMGIyLWMyY2MtNGQzOS1iMTU0LWNiZjkxODU5ZGM0MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyMlQyMDU5MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNTQxNTAwOTBiOTJjYjkzYzgxMTZkYzRiMGE4Y2VjODVmZDk2NDk3NDM2NTkzNjI3MmI1NmU5ZTIzMGMyNjM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Cy9ArMa80i5eA5gDP6p5KHBl5SV4QM_z9h5B8rPTL08"> 

Spacing added to bottom of modal

alexandercyu committed:
Update stac-ui.css

- Fix modal close button highlighting on focus
- Move space in metadata modal from the margin of the interior element to the padding of the container

alexandercyu committed:
Update dropdown-panel.html

- Comment out function to populate stac version for development(prevents error message in console)

alexandercyu committed:
Update dropdown.html

- Attempt to get dropdown background colours switching properly by using the 'show' class put in by bootstrap when the button is clicked
- Add function to count number of checkboxes checked
- Remove unneeded parameters in count checkboxes function
- Change the replace item function in the "Do Not Repeat" dropdown to swap item

alexandercyu committed:
Create dropdown-default-scripts.js

Add functions for default dropdown

alexandercyu committed:
Update dropdown-list.js

Add function to make bottom border of the first item in the small and medium dropdown lists square when they are clicked

alexandercyu committed:
Create dropdown-regular-scripts.js

Add function to change the background colour of the dropdown

alexandercyu committed:
Update sample.html

Add new variable for UL element in dropdown

alexandercyu committed:
Update stac-ui.css

Make placeholder text grey while keeping default black text colour

alexandercyu commented:
<img alt="Screenshot 2025-01-13 at 11 41 41 AM" width="155" src="https://private-user-images.githubusercontent.com/161752175/402622062-ff32d8d6-9514-49de-8475-5d6aca413ad2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc1Nzk4NTIsIm5iZiI6MTczNzU3OTU1MiwicGF0aCI6Ii8xNjE3NTIxNzUvNDAyNjIyMDYyLWZmMzJkOGQ2LTk1MTQtNDlkZS04NDc1LTVkNmFjYTQxM2FkMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyMlQyMDU5MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNWJhZDBmYzliMjhjMDUxMDQ0YTE1OTNkMDQ0ZTYwMGFmODhlYzdiZjk1MzE1ZWVkYjM5YjRhMWQ5Yzk0YjkxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.a1AFpso_n2iESADe0h1Z4C-doGUuN75nKfu2FWOQ5-I"> This button component items need to be center-aligned, <img alt="Screenshot 2025-01-13 at 11 45 18 AM" width="145" src="https://private-user-images.githubusercontent.com/161752175/402623490-a5090783-be91-4783-aaea-d9c81aaee92d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc1Nzk4NTIsIm5iZiI6MTczNzU3OTU1MiwicGF0aCI6Ii8xNjE3NTIxNzUvNDAyNjIzNDkwLWE1MDkwNzgzLWJlOTEtNDc4My1hYWVhLWQ5YzgxYWFlZTkyZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyMlQyMDU5MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZmVhOGUxNTQyMTgwMzM3YjljZWZhZjZlNTc2ZDcwMGY4MjhhZjUwNjViNGUxYmNhOGJkMDdkMjE1NGVkNjVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.x17CAh0SHvkZCBx8jeWPr1S-7ftm2qmYiW2TsWArQDg">

Space between icon and text was actually less than it should have been.  Added more space between icon and text.

alexandercyu commented:
<img alt="Screenshot 2025-01-13 at 11 50 42 AM" width="147" src="https://private-user-images.githubusercontent.com/161752175/402625322-ba143dd1-526e-4a95-b166-32940601e7e9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc1Nzk4NTIsIm5iZiI6MTczNzU3OTU1MiwicGF0aCI6Ii8xNjE3NTIxNzUvNDAyNjI1MzIyLWJhMTQzZGQxLTUyNmUtNGE5NS1iMTY2LTMyOTQwNjAxZTdlOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyMlQyMDU5MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYmJhYTZkMTE3MDcwNjU4ZDhmMjk2YjEzY2U1YjY3MzdhZmYwMDEzZTMxNzQzODI0YmRhYzU0OGVhZDY2NThmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ijjhOe-nc8aLVdzK--LDzhlnaJ9cu9b2BlhZY3N5Vk0">
> 

As mentioned in meeting yesterday, the placeholder text is grey and entered text is black for date and time inputs

alexandercyu committed:
Update stac-ui.css

Test pure css method of swapping background colour and rotating chevron icon in the default dropdown list.

alexandercyu committed:
Update dropdown.html

Add list class back
